File: solutions/python/allergies/1/allergies.py
ALLERGIES = {128:"cats", 64:"pollen", 32:"chocolate", 16:"tomatoes", 8:"strawberries", 4:"shellfish", 2:"peanuts", 1:"eggs"}

class Allergies:

    def __init__(self, score):
        self.list_allergies = []
        next = 256
        while score > 255:
            score -= next
            next *= 2
            
        while score > 0:
            for sco in ALLERGIES:
                if score >= sco:
                    self.list_allergies.insert(0,ALLERGIES[sco])
                    score -= sco

    def allergic_to(self, item):
        if item in self.list_allergies:
            return True
        else:
            return False

    @property
    def lst(self):
        return self.list_allergies

File: solutions/javascript/allergies/1/allergies.js
//
// This is only a SKELETON file for the 'Allergies' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

const ALERGIES = [
  "eggs",
  "peanuts",
  "shellfish",
  "strawberries",
  "tomatoes",
  "chocolate",
  "pollen",
  "cats",
];

export class Allergies {
  constructor(alergias) {
    this.listAlergias = [];
    for (let i = 0; i < ALERGIES.length; i++) {
      if (Number(alergias) & (1 << i)) this.listAlergias.push(ALERGIES[i]);
    }
  }

  list() {
    return this.listAlergias;
  }

  allergicTo(alergie) {
    return this.listAlergias.includes(alergie);
  }
}
